ksql-rest-app/src/test/java/io/confluent/ksql/rest/server/computation/KafkaConfigStoreTest.java

```diff
@@ -0,0 +1,204 @@
+ * limitations under the License.
+ **/
+
+package io.confluent.ksql.rest.server.computation;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyMap;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verifyZeroInteractions;
+import static org.mockito.Mockito.when;
+
+import com.google.common.collect.ImmutableMap;
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+import io.confluent.ksql.rest.server.computation.KafkaConfigStore.KsqlProperties;
+import io.confluent.ksql.rest.util.KsqlInternalTopicUtils;
+import io.confluent.ksql.services.KafkaTopicClient;
+import io.confluent.ksql.util.KsqlConfig;
+import java.util.LinkedList;
+import java.util.function.Supplier;
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.clients.consumer.ConsumerRecords;
+import org.apache.kafka.clients.consumer.KafkaConsumer;
+import org.apache.kafka.clients.producer.KafkaProducer;
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.apache.kafka.common.TopicPartition;
+import org.apache.kafka.common.serialization.Deserializer;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+
+import java.util.Collections;
+import java.util.List;
+import org.mockito.ArgumentCaptor;
+import org.mockito.InOrder;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.MockitoJUnit;
+import org.mockito.junit.MockitoRule;
+
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.junit.Assert.assertThat;
+
+public class KafkaConfigStoreTest {
+  private final KsqlConfig ksqlConfig = new KsqlConfig(
+      ImmutableMap.of(
+        KsqlConfig.KSQL_SERVICE_ID_CONFIG, "test-service-id"
+      )
+  );
+  private final String topicName = KsqlInternalTopicUtils.getTopicName(ksqlConfig, "configs");
+  private final TopicPartition topicPartition = new TopicPartition(topicName, 0);
+  private final List<TopicPartition> topicPartitionAsList
+      = Collections.singletonList(topicPartition);
+
+  @Mock
+  private Supplier<KafkaConsumer<String, KsqlProperties>> consumerSupplier;
+  @Mock
+  private Supplier<KafkaProducer<String, KsqlProperties>> producerSupplier;
+  @Mock
+  private KafkaProducer<String, KsqlProperties> producer;
+  @Mock
+  private KafkaTopicClient topicClient;
+
+  private KafkaConfigStore configStore;
+
+  @Rule
+  public final MockitoRule mockitoRule = MockitoJUnit.rule();
+
+  @Before
+  @SuppressWarnings("unchecked")
+  public void setUp() {
+    when(topicClient.isTopicExists(anyString())).thenReturn(true);
+    when(topicClient.addTopicConfig(anyString(), anyMap()))
+        .thenReturn(false);
+    when(producerSupplier.get()).thenReturn(producer);
+  }
+
+  private void initConfigStore() {
+    configStore = new KafkaConfigStore(ksqlConfig, topicClient, consumerSupplier, producerSupplier);
```

Comment:
this can just be done in `setUp` right?

Reply:
KafkaConfigStore reads/writes the config from its constructor